src/integ_test_resources/ios/sdk/integration/cdk/custom_resources/iot_custom_authorizerUserPass_provider/iot_custom_authorizerUserPass_provider.py

```diff
@@ -0,0 +1,107 @@
+import boto3
+
+# Note: Make this dynamic if there is more than one custom authorizer in the stack
+physical_id = "iot_custom_authorizerUserPass_provider_20200508074521"
+domain_configuration_name = "aws_test_iot_custom_authorizerUserPass"
+default_authorizer_name = "iot_custom_authorizerUserPass"
+
+def on_event(event, __):
+    print(f"### on_event({event})")
+    request_type = event["RequestType"]
+    if request_type == "Create":
+        return on_create_or_update(event)
+    if request_type == "Update":
+        return on_create_or_update(event)
+    if request_type == "Delete":
+        return on_delete(event)
+    raise Exception("Invalid request type: %s" % request_type)
+
+
+def on_create_or_update(event):
+    print(f"### on_create_or_update({event})")
+    custom_authorizer_function_arn = event["ResourceProperties"]["authorizer_function_arn"]
+    custom_authorizer_name = event["ResourceProperties"]["authorizer_name"]
+    public_key = event["ResourceProperties"]["public_key"]
+    token_key_name = event["ResourceProperties"]["token_key_name"]
+    create_authorizer(
+        custom_authorizer_name, custom_authorizer_function_arn, public_key, token_key_name
+    )
+
+    #Creating a domain config is required for enhanced user auth
+    beta_endpoint_addr = create_or_update_domainConfig()
+    response = {
+        "PhysicalResourceId": physical_id,
+        "Data": {"BetaEndpointAddress": beta_endpoint_addr}
+    }
+    print(f"### on_create_or_update response: {response}")
+    return response
+
+def create_or_update_domainConfig():
```

Comment:
Ahh. This is the "enhanced" part, right?

Reply:
Correct! This is what is required to enable the enhanced payload as dictated by the service team.  I believe that at some point in time (perhaps once this feature comes out of beta??), this will not be needed.  I believe this was done purely for beta purposes.